internal/pkg/agent/install/uninstall.go

```diff
@@ -290,3 +297,36 @@ func applyDynamics(ctx context.Context, log *logger.Logger, cfg *config.Config)
 
 	return config.NewConfigFrom(finalConfig)
 }
+
+// killWatcher finds and kills any running Elastic Agent watcher.
+func killWatcher() error {
+	for {
```

Comment:
[Blocker]
I don't see a stop condition, when will the function return?
Shouldn't you remove the PIDs which `proc.Kill()` return `os.ErrProcessDone` from the list?

Reply:
`utils.GetWatcherPIDs()` returns a list of PID's. It returns when there are no more found watcher processes.

Reply:
I had this thought too, the only other option is to add a timeout but we have no idea how long to set it to. We could pick some arbitrarily long value beyond which we think users might consider uninstall to be hanging? That's probably 30s to 1 minute. Part of why I wanted logging was so we could see how long this takes in practice.

I suppose this could be tied into the root context of the application so that SIGTERM causes it to stop, if it doesn't already.